### PR TITLE
chore(deps): update dependency joi to v17.3.0

### PR DESCRIPTION
**Pull request by donbot-dev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe:
```

Dependency update

## What is the current behavior?
Screen Shot 2020-11-09 at 3 20 52 PM

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Joi package moved from `@hapi/joi` to `joi` and now includes types.

Also look PR https://github.com/nestjs/docs.nestjs.com/pull/1535 for docs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,12 +27,10 @@
   "devDependencies": {
     "@commitlint/cli": "11.0.0",
     "@commitlint/config-angular": "11.0.0",
-    "@hapi/joi": "17.1.1",
     "@nestjs/common": "7.5.4",
     "@nestjs/core": "7.5.4",
     "@nestjs/platform-express": "7.5.4",
     "@nestjs/testing": "7.5.4",
-    "@types/hapi__joi": "17.1.6",
```
Comment (Keimeno) [inline package.json:35]: You're missing the `@types/joi` package

**Review by Keimeno — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (donbot-dev) [reply package.json:35]: Package `joi` already include typings, or i don't understand something?

Comment (Keimeno) [reply package.json:35]: My bad sorry, we should get [@types/joi](https://www.npmjs.com/package/@types/joi) deprecated then.

**Review by Keimeno — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**